beacon-chain/blockchain/receive_block.go

```diff
@@ -224,6 +219,17 @@ func (s *Service) ReceiveBlock(ctx context.Context, block interfaces.ReadOnlySig
 	return nil
 }
 
+func (s *Service) executePostFinalizationTasks(ctx context.Context, finalizedState state.BeaconState) {
+	finalized := s.cfg.ForkChoiceStore.FinalizedCheckpoint()
+	finalizedState.SaveValidatorIndices() // used to handle Validator index invariant from EIP6110
```

Comment:
Do we need to block here? Should `SaveValidatorIndices` be inside `go func()`?

Reply:
I can place it with the send New finalized event